### DISCUSSION TITLE
Good first issues?

### DISCUSSION THREAD

**Discussion by jsimonrichard** [GitoxideLabs/gitoxide#1409] [Ideas]
I'm still wrapping my head around this repo, but I'd be happy to help out if I can. I'm working on a rust project that will require several of the uncompleted features mentioned in #1287 and #1315 including...
* [ ] `git status` (especially the status of staged files)
* [ ] `git add`
* [ ] Reset
* [ ] Merging
* [ ] Signed commits

Along with probably most notably:
* [ ] `git push`

Are there any good first issues, especially considering these features? I don't have the knowledge to implement any of these features right away, but if there's anything helpful that I could do to move `gix` closer to implementing any of these features, please let me know.

I'm working on a startup that depends on async git rust tooling; I may have to switch over to using the git CLI (like `gitu`) for now, but I'd like to use `gix` as much as possible.

**Comment by Byron** [comment]
Thanks for getting involved! 

The first task on [this PR](https://github.com/Byron/gitoxide/pull/1368) is an index-index diff which is the last part needed to get `status` to work. Git might do an actual tree-index diff, but in theory it should be easier to do index-index first as an index can conveniently be created from a tree.
Devil is in the details though, and the index can have entries related to sparse checkouts which need some care to be taken.

For the last two months I didn't get to do any feature work, but I am trying to catchup. Maybe you can get started on that in your own PR which I recommend to bring up in draft form as early as the first line of code so feedback is possible early as well.
